test/color/test_zca.py

```diff
@@ -0,0 +1,165 @@
+
+    def test_grad_zca_individual_transforms(self, device):
+        """
+
+        Checks if the gradients of the transforms are correct w.r.t to the input data
+
+        """
+
+        data = torch.tensor([[2, 0],
+                             [0, 1],
+                             [-2, 0],
+                             [0, -1]],
+                            dtype=torch.float32).to(device)
+
+        data = utils.tensor_to_gradcheck_var(data)
+
+        def zca_T(x):
+            return kornia.color.zca_mean(x)[0]
+
+        def zca_mu(x):
+            return kornia.color.zca_mean(x)[1]
+
+        def zca_T_inv(x):
+            return kornia.color.zca_mean(x, compute_inv=True)[2]
+
+        assert gradcheck(zca_T, (data,), raise_exception=True)
+        assert gradcheck(zca_mu, (data,), raise_exception=True)
+        assert gradcheck(zca_T_inv, (data,), raise_exception=True)
+
+    def test_grad_zca_with_fit(self, device):
+
+        data = torch.tensor([[2, 0],
+                             [0, 1],
+                             [-2, 0],
+                             [0, -1]],
+                            dtype=torch.float32).to(device)
+
+        data = utils.tensor_to_gradcheck_var(data)
+
+        def zca_fit(x):
+            zca = kornia.color.ZCAWhitening(detach_transforms=False)
+            return zca(x, include_fit=True)
+
+        assert gradcheck(zca_fit, (data,), raise_exception=True)
+
+    def test_grad_detach_zca(self, device):
+
+        data = torch.tensor([[2, 0],
+                             [0, 1],
+                             [-2, 0],
+                             [0, -1]],
+                            dtype=torch.float32).to(device)
+
+        data = utils.tensor_to_gradcheck_var(data)
+        zca = kornia.color.ZCAWhitening(detach_transforms=True).fit(data)
+
+        assert gradcheck(zca,
+                         (data,), raise_exception=True)
+
+    def test_not_fitted(self, device):
+
+        with pytest.raises(RuntimeError):
+            data = torch.rand(10, 2).to(device)
+
+            zca = kornia.color.ZCAWhitening()
+            zca(data)
+
+    def test_not_fitted_inv(self, device):
+
+        with pytest.raises(RuntimeError):
+            data = torch.rand(10, 2).to(device)
+
+            zca = kornia.color.ZCAWhitening()
+            zca.inverse_transform(data)
+
+    def test_with_linear_transform(self, device):
+        data = torch.tensor([[1, 0],
+                             [0, 1],
+                             [-1, 0],
+                             [0, -1]],
+                            dtype=torch.float32).to(device)
+        data = data.view(4, 1, 2, 1)
+        expected = math.sqrt(3 / 2) * data
+
+        T, mu = kornia.color.zca_mean(data)
+
+        lt = LinearTransformation(T, mu)
+
+        out = torch.stack([lt(data[0]), lt(data[1]), lt(data[2]), lt(data[3])], axis=0)
+
+        assert_allclose(out, expected)
+
+    @pytest.mark.skip(reason="turn off all jit for a while")
```

Comment:
Yeah. I'll try this out.

Reply:
Hey @edgarriba , I was taking a look at the jit stuff. It seems ```torch.is_tensor``` is problematic for ```torch.jit.script```. I added a [Colab Notebook](https://colab.research.google.com/drive/1stYuhcoRTradbRm9HqJlcBUZqJXYnier) showing the problem. It's odd since  ```torch.is_tensor```  is essentially ```isinstance(x, torch.Tensor)``` according to [here](https://pytorch.org/docs/stable/_modules/torch.html#is_tensor)

Reply:
you are right :)
actually JIT author suggested to use isinstance since would be more pythonic, otherwise we can send a PR directly to cover this ops in JIT. Would go for pythonic way.

Besides, I have a pending work to create a minimal API in kornia.testing to have different `check_*` functions. `check_is_tensor` would be the first one in the list :)

Reply:
`check_type` could be another one that receives multiple possible input type